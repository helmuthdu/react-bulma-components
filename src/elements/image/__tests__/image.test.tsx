import { fireEvent, render, waitFor } from '@testing-library/react';
import * as React from 'react';
import { Image } from '..';

describe('Image component', () => {
  it('should render', () => {
    const { asFragment } = render(<Image src="http://mydomain.com/image" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should be rounded', () => {
    const { asFragment } = render(<Image rounded src="http://mydomain.com/image" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should be 32x32', () => {
    const { asFragment } = render(<Image size={32} src="http://mydomain.com/image" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have use default image if error encounter', async () => {
    const { container } = render(<Image src="http://mydomain.com/image" fallbackSrc="http://mydomain.com/default" />);
    const image = container.querySelector('img') as HTMLImageElement;
    fireEvent.error(image);

    await waitFor(() => {
      expect(image).toHaveAttribute('src', 'http://mydomain.com/default');
    });
  });

  it('should update src', async () => {
    const getComponent = (url: string) => <Image src={url} />;
    const { container, rerender } = render(getComponent('http://mydomain.com/image'));
    rerender(getComponent('http://mydomain.com/other'));
    const image = container.querySelector('img') as HTMLImageElement;
    expect(image).toHaveAttribute('src', 'http://mydomain.com/other');
  });
});
