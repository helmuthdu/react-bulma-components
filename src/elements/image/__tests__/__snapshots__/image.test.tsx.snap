// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component should be 32x32 1`] = `
<DocumentFragment>
  <figure
    class="image is-32x32"
  >
    <img
      class=""
      src="http://mydomain.com/image"
    />
  </figure>
</DocumentFragment>
`;

exports[`Image component should be rounded 1`] = `
<DocumentFragment>
  <figure
    class="image"
  >
    <img
      class="is-rounded"
      src="http://mydomain.com/image"
    />
  </figure>
</DocumentFragment>
`;

exports[`Image component should render 1`] = `
<DocumentFragment>
  <figure
    class="image"
  >
    <img
      class=""
      src="http://mydomain.com/image"
    />
  </figure>
</DocumentFragment>
`;
